### ISSUE TITLE
Seems to work for me

### ISSUE THREAD

**Issue by inspirit** [lucidrains/grokfast-pytorch#1]
Hi Phil, I tested it in my private project 2 days ago, and it seems to speed up learning quite significantly, not sure that final val/train losses are better, more like very similar to original but it got there much faster. Also i did not do different tasks/architects to compare, but my project contains few different nets one including tiny transformer, another using RNN cells and last one simple shallow convolutions

**Comment by lucidrains** [comment]
hi Eugene and thanks for reporting this!

did you by chance compare it with Adam but with half the learning rate? I don't think it is a fair comparison as they effectively doubled their learning rate by summing the slow grads

**Comment by inspirit** [comment]
nope i did not change any parameters of Adam opt, I just added ema grads step between loss.backward() and opt.step() as mentioned in the paper, so its indeed not fair comparison :)

**Comment by lucidrains** [comment]
how big was the speed up you saw?

**Comment by inspirit** [comment]
25-30% faster

**Comment by lucidrains** [comment]
yeah that isn't significant for double the lr, imo

**Comment by lucidrains** [comment]
Ill try it for some other tasks other than modulus addition

**Comment by lucidrains** [comment]
actually, it is more like 3x the lr given the `lamb` they use is `2`

**Comment by inspirit** [comment]
yup, thats rights

**Comment by inspirit** [comment]
Looks like i was celebrating too early, 
comparing now with Lion: after some trials searching for correct LR i can see that grokfast better right from the start for about 3 epochs afterwards lion catches up

**Comment by lucidrains** [comment]
@inspirit thanks! was the comparison with lion + grokfast with lion?

what kind of task are you training on?

**Comment by inspirit** [comment]
it was grokfast+adam vs lion, i have a recurrent generator net with several conditioning encoders for audio and text.
what i like about grokfast is that its quite stable with almost any optimizer configuration, while lion you need to tune to make it work

**Comment by lucidrains** [comment]
@inspirit yea, you should control for the same optimizer at least

but it is safe to say at this point that grokfast doesn't seem to produce any immediate improvements for practical tasks out of the box. i'm still interested in these tiny algorithmic tasks though; i'm wondering if one can curriculum grok a series of N tasks

**Comment by lucidrains** [comment]
yea we'll see, this whole thing could be a dud lol

**Comment by inspirit** [comment]
it seems to be very tricky to make it work with Lion opt, did not find LR combination that make it learn with grokfast
